### DISCUSSION TITLE
json_schema_extra not supported in Field

### DISCUSSION THREAD

**Discussion by palfrey** [fastapi/sqlmodel#780] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import Field

# This should work, according to Pydantic docs
class PydanticHero(SQLModel):
    foo: str = Field(json_schema_extra={ 'enum': ["some", "values"})

# This currently does work as a hacky workaround
class SQLModelHero(SQLModel):
    foo: str = Field(schema_extra={"json_schema_extra":{ 'enum': ["some", "values"}})
```

### Description

I've got a case where I want to set some valid enum values in a schema, and this should be doable with `json_schema_extra`. Except it's a Pydantic v2 feature, and SQLModel only seems to support the v1 version (see also https://docs.pydantic.dev/latest/migration/#changes-to-pydanticfield). I found a hacky workaround (see example), but it's kinda terrible

### Operating System

macOS

### SQLModel Version

0.0.14

### Python Version

3.10

### Additional Context

Running pydantic 2.5.3

**Comment by DirkReiners-UCF** [comment]
I have some other uses for json_schema_extra, and right now this is preventing me from moving forward with SQLModel... :(

**Comment by ahaly-compass** [comment]
Any update on this?

**Comment by OliverEvans96** [comment]
This other answer has a workaround using both SQLModel's schema_extra and Pydantic's json_schema_extra

https://github.com/fastapi/sqlmodel/discussions/833#discussioncomment-9366092
